src/ftest.jl

```diff
@@ -0,0 +1,84 @@
+type FTestResult
+    SSR1::Float64 # sum of squared residuals, model 1
+    SSR2::Float64 # sum of squared residuals, model 2
+    df1::Int # degrees of freedom, model 1 residuals
+    df2::Int # degrees of freedom, model 2 residuals
+    MSR1::Float64 # mean of squared residuals, model 1
+    MSR2::Float64 # mean of squared residuals, model 2
+    fstat::Float64 # f statistic
+    pval::Float64 # p value
+end
+
+type MultiFTestResult
+    results::Array{FTestResult, 1}
+end
+
+function ftest(mod1::LinPredModel, mod2::LinPredModel)
+    SSR1 = deviance(mod1)
+    SSR2 = deviance(mod2)
+
+    nparams1 = dof(mod1)
+    nparams2 = dof(mod2)
+
+    df2 = nparams1-nparams2
+    df1 = dof_residual(mod1)
+
+    MSR1 = (SSR2-SSR1)/df2
+    MSR2 = SSR1/df1
+
+    fstat = MSR1/MSR2
+    pval = ccdf(FDist(df2, df1), fstat)
+
+    return FTestResult(SSR1, SSR2, df1, df2, MSR1, MSR2, fstat, pval)
+end
+
+function ftest(mods::LinPredModel...)
+    nmodels = length(mods)
+    SSR1s = Array{FTestResult, 1}(nmodels-1)
+    SSR2s = similar(SSR1s)
+    df1s = similar(SSR1s)
+    df2s = similar(SSR1s)
+    MSR1s = similar(SSR1s)
+    MSR2s = similar(SSR1s)
+    fstats = similar(SSR1s)
+    pvals = similar(SSR1s)
+    
+    for (i, mod1) in enumerate(mods[2:end])
+        result = ftest(mods[i], mod1)
+        SSR1s[i] = result.SSR1
+        SSR2s[i] = result.SSR2
+        df1s[i] = result.df1
+        df2s[i] = result.df2
+        MSR1s[i] = result.MSR1
+        MSR2s[i] = result.MSR2
+        fstats[i] = result.fstat
+        pvals[i] = result.pval
+    end
+
+    return CoefTable([SSR1s, SSR2s, df1s, df2s, MSR1s, MSR2s, fstats, pvals],
```

Comment:
This isn't an opinion, but rather an explanation. I defined FTestResult purely so that I could define show() for it and have the result nicely printed.

Is there some more standard way to do this?